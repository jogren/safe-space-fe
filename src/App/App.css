* {
  box-sizing: border-box;
}

.App {
  background-color: #6C782A;
  height: 100vh;
}