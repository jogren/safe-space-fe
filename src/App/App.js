import React, { useState, useMemo } from 'react'
import './App.css'
import Header from '../Header/Header'
import SplashPage from '../SplashPage/SplashPage'
import LogInForm from '../LogInForm/LogInForm'
import CheckInForm from "../CheckInForm/CheckInForm";
import CheckOutForm from "../CheckOutForm/CheckOutForm";
import { Route } from "react-router-dom";
import { ApolloProvider } from '@apollo/react-hooks';
import ApolloClient from 'apollo-boost';
import { CurrentCenterContext } from '../CurrentCenterContext';
const client = new ApolloClient({
  uri: 'https://safe-space-be.herokuapp.com/graphql',
});

const App = () => {
  const [reliefCenter, setreliefCenter] = useState('');
  const [isCenterSelected, selectCenter] = useState(false)
  const [isLoading, setIsLoading] = useState(false);

  const currentReliefCenter = useMemo(() => ({ reliefCenter, setreliefCenter }), [reliefCenter, setreliefCenter]);
    
    //add componentDidMount to fetch relief center data
    // once fetched, set isLoading to false
    
    return (
      <ApolloProvider client={client}>
      <section className="App">
        <Header />
        <CurrentCenterContext.Provider value={currentReliefCenter}>
          <Route exact path="/">
            {isLoading ? (
              <SplashPage />
              ) : (
                <LogInForm
                selectCenter={selectCenter}
                isCenterSelected={isCenterSelected}
                reliefCenter={reliefCenter}
                />
                )}
          </Route>
          {/* <Route exact path='/supplies' render={() => < /> */}
          <Route exact path="/check-in" component={CheckInForm} />
          <Route exact path="/check-out" component={CheckOutForm} />
        </CurrentCenterContext.Provider>
      </section>
    </ApolloProvider>
  );
}

export default App;
