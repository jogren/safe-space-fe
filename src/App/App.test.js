import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';


describe('Header', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  it('should match the snapshot with the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  })
})