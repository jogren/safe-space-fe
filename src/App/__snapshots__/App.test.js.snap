// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot with the data passed through 1`] = `undefined`;
